extension/js/common/core/crypto/key.ts

```diff
@@ -297,4 +297,22 @@ export class KeyUtil {
     }
   }
 
+  public static keyInfoObj = async (prv: Key, passphrase?: string): Promise<KeyInfo> => {
```

Comment:
Looking at KeyInfo and PrvKeyInfo, I'm not sure if there is any use to having both anymore. I think nearly all usecases we have are for `PrvKeyInfo`. Wherever we need `KeyInfo`, we should use a `Key` instead (without storing it in storage).

I'm not 100% sure about this, but I think we could simplify this way.

Reply:
`public` property is read in many places, for example, in keyserver (`replacePublicKeyHandler`)
In other places, like composer module, do you suggest to replace parse(ki.public) with parse(ki.private).asPublicKey?

Reply:
I'll have to have a look and maybe attempt refactoring that away. I suspect there are a bunch of usages that are no longer needed, but I'll have to see them one by one to confirm.